t/re/reg_mesg.t

```diff
@@ -20,6 +20,25 @@ use open qw(:utf8 :std);
 use warnings;
 BEGIN { ${^WARNING_BITS} = undef }  # Kludge to restore default warnings
 
+# this is used to output the patterns we are testing, it is specifically
+# written to not use any regex logic. Currently it assumes that we are
+# using ASCII/Unicode encoding. Maybe one day we will extend it to
+# handle EBCDIC, although I am not sure it is necessary. - Yves
+sub escape_string {
+    my ($str) = @_;
+    my $escaped = "";
+    for my $i (0 .. length($str)-1) {
+        my $char = substr($str,$i,1);
+        my $ord = ord($char);
+        if (32 <= $ord <= 127) {
```

Comment:
Should be `126` (or `< 127`) for printable ASCII.

Reply:
Good catch. I changed the end range to 126 as you suggested, but I'm not sure why we need the second part with `"\\"` as it is codepoint  92 (0x5c) , so it shouldn't be escaped anyway.

Reply:
In case you want to distinguish `escape_string("\\\0")` from `escape_string("\\x{00}")`. Currently they produce `\\x{00}`.

Reply:
Regarding the backslash, did you mean to escape backslashes anyway? I guess at a certain level that makes sense, but  if we go down that road maybe we should do what the regex debugger does and use % instead of backslash so that actual backslashes in the input stay as backslashes. I doubt if any of our tests include a percentage symbol. Having said that I doubt its necessary, we just need enough info to work out which test failed.

Reply:
oh weird, the page didnt update automatically, so i didnt see your most recent comment. I guess ill change it to use %.

Reply:
Ok, i just updated it. `escape_string("\\\0")` should output `\%x{00}` and escape_string("\\x{00}") should output `\x{00}`